src/algebra/monoid_algebra.lean

```diff
@@ -571,6 +591,25 @@ begin
   simp only [finsupp.sum_apply, single_apply],
 end
 
+lemma mul_apply_antidiagonal (f g : add_monoid_algebra k G) (x : G) (s : finset (G × G))
+  (hs : ∀ {p : G × G}, p ∈ s ↔ p.1 + p.2 = x) :
+  (f * g) x = ∑ p in s, (f p.1 * g p.2) :=
+let F : G × G → k := λ p, if p.1 + p.2 = x then f p.1 * g p.2 else 0 in
+calc (f * g) x = (∑ a₁ in f.support, ∑ a₂ in g.support, F (a₁, a₂)) :
+  mul_apply f g x
+... = ∑ p in f.support.product g.support, F p : finset.sum_product.symm
+... = ∑ p in (f.support.product g.support).filter (λ p : G × G, p.1 + p.2 = x), f p.1 * g p.2 :
+  (finset.sum_filter _ _).symm
+... = ∑ p in s.filter (λ p : G × G, p.1 ∈ f.support ∧ p.2 ∈ g.support), f p.1 * g p.2 :
+  sum_congr (by { ext, simp only [mem_filter, mem_product, hs, and_comm] }) (λ _ _, rfl)
+... = ∑ p in s, f p.1 * g p.2 : sum_subset (filter_subset _) $ λ p hps hp,
+  begin
+    simp only [mem_filter, mem_support_iff, not_and, not_not] at hp ⊢,
+    by_cases h1 : f p.1 = 0,
+    { rw [h1, zero_mul] },
+    { rw [hp hps h1, mul_zero] }
+  end
```

Comment:
As noted in the commit messaeg, this is just `monoid_algebra.mul_apply_antidiagonal` with a couple `*` replaced with `+`.